lib/purl_listener.rb

```diff
@@ -0,0 +1,179 @@
+require 'benchmark'
+require 'fileutils'
+require 'listen'
+
+##
+# PurlListener supports incremental PURL updates via a "dropbox" folder
+# on the PURL filesystem. The implementation assumes that the PURL filesystem
+# is mounted via NFS and is a read-write mount.
+#
+# This is the basic algorithm:
+#
+# watches "dropbox" directory for filesystem events,
+# then on event:
+#     - logs that the listener process is active
+#     - rename `aa111bb2222` to `aa111bb2222.lock`
+#     - save to `Purl` model, and logs (to file and Honeybadger) any exceptions (then continues)
+#     - delete `aa111bb2222.lock`
+#
+# when listener boots:
+#     - it checks to see when it was last active
+#     - then runs a `find:all[mins_ago]` task
+#
+class PurlListener
+
+  attr_reader :event_handler, :logger, :path, :pid_file
+
+  ##
+  # @param [Pathname] `path` -- the single "dropbox" directory
+  # @param [Pathname] `pid_file` -- the location of the pid file
+  # @param [Logger] `logger` -- place to log info/warn/errors
+  # @raise [ArgumentError] -- requires path to be a directory
+  def initialize(path = Pathname(PurlFetcher::Application.config.app_config['listener_path']),
+                 pid_file = Pathname(PurlFetcher::Application.config.app_config['listener_pid_file']),
+                 logger = IndexingLogger)
```

Comment:
Yes, `IndexingLogger` goes to the `log/indexing.log` file where as Rails logger goes to `log/production.log`. I think the idea is to keep all the "indexing" activity distinct from the API activity.